interface/billing/search_payments.php

```diff
@@ -51,7 +57,8 @@
                 sqlStatement("update form_encounter set last_level_closed=last_level_closed - 1 where pid =? and encounter=?", [$PId, $Encounter]);
             }
         }
-
+        //dispatch this payment is being deleted trigger refund process
+        $eventDispatcher->dispatch(new DeletePayment(add_escape_custom($DeletePaymentId)), DeletePayment::ACTION_DELETE_PAYMENT, 10);
```

Comment:
I wouldn't escape the payment id here.  Downstream consumers will then have to remove the escaping it if they use it for something different than database insertion.

Reply:
I understand what you are saying, and in my use case, that'd be great. I only use the ID to retrieve information once I determine that the credit card processor makes the entry in the database.